packages/flutter_tools/test/src/testbed.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/context_runner.dart';
+
+export 'package:flutter_tools/src/base/context.dart' show Generator;
+
+final Map<Type, Generator> _testbedDefaults = <Type, Generator>{
+  FileSystem: () => MemoryFileSystem(),
+  Logger: () => BufferLogger(),
+};
+
+/// Manages interaction with the tool injection and runner system.
```

Comment:
Maybe add example usage to the comments, or point at a simple reference example.

Reply:
Added an example and more documentation